### PR TITLE
Bug 1469369 Messages get stuck after changing configuration

### PR DESCRIPTION
**Pull request by gaohoward**
If user changes max-size-bytes from a limited value to -1 and
restart the broker, the paged messages in storage will get
stuck after restarting. This is because that the broker
doesn't do depaging when max-size-bytes is -1.

To fix it, just makes the broke do a depage even if
max-size-bytes is -1.

### CODE REVIEW

File: hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -2181,7 +2180,7 @@ private void checkDepage()
     */
    private boolean needsDepage()
    {
-      return queueMemorySize.get() < pageSubscription.getPagingStore().getMaxSize();
+      return queueMemorySize.get() < pageSubscription.getPagingStore().getMaxSize() || pageSubscription.getPagingStore().getMaxSize() == -1;
```
Comment (clebertsuconic) [inline hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java:2183]: Artemis?

Comment (clebertsuconic) [reply hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java:2183]: I don't want to merge fixes in HornetQ if they don't have an equivalent fix in Artemis... it would get out of control easily.

Comment (gaohoward) [reply hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java:2183]: @clebertsuconic I investigated the Artemis case and found that it doesn't have this kind of issue anymore. It happens that Artemis fixes this issue by a commit fixing Artemis-581. The part that fixes this particular issue is in PagingStoreImpl.java:

    @Override
        public long getMaxSize() {
    -      return maxSize;
    +      if (maxSize < 0) {
    +         // if maxSize < 0, we will return 2 pages for depage purposes
    +         return pageSize * 2;
    +      }
    +      else {
    +         return maxSize;
    +      }

I'm not sure what's the purpose for fixing Artemis-581, but this avoids this HornetQ issue.
So basically Artemis don't need the fix. Only I think the test is valid and can be ported into Artemis.

Howard

Comment (gaohoward) [reply hornetq-server/src/main/java/org/hornetq/core/server/impl/QueueImpl.java:2183]: @clebertsuconic Can you merge this PR pls?
